libs/testserver/server.go

```diff
@@ -33,11 +34,42 @@ func New(t testutil.TestingT) *Server {
 	server := httptest.NewServer(mux)
 	t.Cleanup(server.Close)
 
-	return &Server{
+	s := &Server{
 		Server: server,
 		Mux:    mux,
 		t:      t,
 	}
+
+	// The server resolves conflicting handlers by using the one with higher
+	// specificity. This handler is the least specific, so it will be used as a
+	// fallback when no other handlers match.
+	s.Handle("/", func(r *http.Request) (any, int) {
+		pattern := r.Method + " " + r.URL.Path
+
+		t.Errorf(`
+
+----------------------------------------
+No stub found for pattern: %s
+
+To stub a response for this request, you can add
+the following to test.toml:
+[[Server]]
+Pattern = %q
+Response.Body = '''
+<response body here>
+'''
+Response.StatusCode = <response status-code here>
```

Comment:
If we do error, can we keep it short? It would be hard to read if all tests start complaining about the same missing API. Feel free to add section about this to acceptance/README.

However, we should clarify what's the motivation for this change.

Reply:
My hunch is to make copy/paste of missing handlers easier. @shreyas-goenka can you confirm?

Reply:
Yeah, that's exactly right. It's there to make copy-pasting easier.

Reply:
Please clarify this and include the motivation for erroring in the PR summary.

Reply:
Please include this and include the motivation for erroring in the PR summary.